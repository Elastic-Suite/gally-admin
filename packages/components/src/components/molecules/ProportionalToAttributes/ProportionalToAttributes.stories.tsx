import React, { useState } from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import ProportionalToAttributesComponent, {
  IProportionalToAttributesValue,
} from './ProportionalToAttributes'
import boostImpactOptionsMock from '../../../../public/mocks/boost_impact_options.json'

export default {
  title: 'Molecules/ProportionalToAttributesComponent',
  component: ProportionalToAttributesComponent,
} as ComponentMeta<typeof ProportionalToAttributesComponent>

const Template: ComponentStory<typeof ProportionalToAttributesComponent> = (
  args
) => {
  const [value, setValue] = useState<IProportionalToAttributesValue>({
    source_field_code: undefined,
    boost_impact: undefined,
    scale_factor: 1,
  })
  return (
    <ProportionalToAttributesComponent
      {...args}
      value={value}
      onChange={(value): void => setValue(value)}
    />
  )
}

export const ProportionalToAttributes = Template.bind({})

ProportionalToAttributes.args = {
  boostImpactOptions: boostImpactOptionsMock['hydra:member'],
  sourceFields: [
    {
      label: 'test',
      value: 'test',
    },
  ],
}
