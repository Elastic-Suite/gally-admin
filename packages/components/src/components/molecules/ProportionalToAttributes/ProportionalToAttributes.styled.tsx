import { styled } from '@mui/material'

export const ProportionalToAttributesContainer = styled('div')(({ theme }) => ({
  display: 'flex',
  flexDirection: 'column',
  gap: theme.spacing(1),
}))
