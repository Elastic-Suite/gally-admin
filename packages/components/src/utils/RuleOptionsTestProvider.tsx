import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
