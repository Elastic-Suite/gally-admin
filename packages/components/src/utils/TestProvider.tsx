import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
