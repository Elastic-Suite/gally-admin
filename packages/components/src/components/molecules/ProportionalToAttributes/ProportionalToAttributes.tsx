import React from 'react'
import RadioGroup from '../../atoms/form/RadioGroup'
import InputText from '../../atoms/form/InputText'
import DropDown from '../../atoms/form/DropDown'
import { IOptions } from '@elastic-suite/gally-admin-shared'
import { useTranslation } from 'next-i18next'
import { ProportionalToAttributesContainer } from './ProportionalToAttributes.styled'
export interface IProportionalToAttributesValue {
  source_field_code?: string
  boost_impact?: string
  scale_factor: number
}

export interface IProportionalToAttributesProps {
  sourceFields: IOptions<string>
  boostImpactOptions: IOptions<string>
  value: IProportionalToAttributesValue
  showError?: boolean
  onChange?: (value: IProportionalToAttributesValue) => void
}

function ProportionalToAttributes({
  value,
  sourceFields,
  boostImpactOptions,
  showError,
  onChange,
}: IProportionalToAttributesProps): JSX.Element {
  const {
    source_field_code: sourceFieldCode,
    boost_impact: boostImpact,
    scale_factor: scaleFactor,
  } = value
  const { t } = useTranslation('boost')

  function handleChange(
    name: keyof IProportionalToAttributesValue,
    newValue: string | number
  ): void {
    if (onChange)
      onChange({
        ...value,
        [name]: newValue,
      })
  }

  return (
    <ProportionalToAttributesContainer>
      <DropDown
        options={sourceFields}
        label={t('boostConfig.sourceField')}
        value={sourceFieldCode}
        onChange={(value: string): void =>
          handleChange('source_field_code', value)
        }
        required
        showError={showError}
        sx={{ minWidth: '230px' }}
        placeholder={t('boostConfig.sourceField.placeholder')}
        fullWidth
      />

      <RadioGroup
        options={boostImpactOptions}
        value={boostImpact || null}
        onChange={(value: string): void => handleChange('boost_impact', value)}
        label={t('boostConfig.boostImpact')}
        required
        showError={showError}
      />

      <InputText
        type="number"
        label={t('boostConfig.scaleFactor')}
        required
        inputProps={{
          step: 'any',
          min: 0,
        }}
        additionalValidator={(value: number): string => {
          if (value <= 0) {
            return 'scaleFactorWrongFormat'
          }
          return ''
        }}
        value={scaleFactor}
        onChange={(value: number): void => handleChange('scale_factor', value)}
        showError={showError}
      />
    </ProportionalToAttributesContainer>
  )
}

export default ProportionalToAttributes
