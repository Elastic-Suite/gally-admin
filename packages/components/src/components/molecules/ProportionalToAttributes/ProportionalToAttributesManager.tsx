import React from 'react'
import { useApiList, useResource } from '../../../hooks'
import ProportionalToAttributes, {
  IProportionalToAttributesProps,
} from './ProportionalToAttributes'
import { IOption } from '@elastic-suite/gally-admin-shared'

const rowsPerPage = 200
const sourceFieldFixedFilters = {
  'metadata.entity': 'product',
}

function ProportionalToAttributesManager(
  props: Omit<
    IProportionalToAttributesProps,
    'sourceFields' | 'boostImpactOptions'
  >
): JSX.Element {
  const boostImpactOptionResource = useResource('BoostImpactOption')
  const boostAttributeValueFieldOptionsResource = useResource(
    'BoostAttributeValueFieldOption'
  )

  const [boostImpactOptionsResponse] = useApiList<IOption<string>>(
    boostImpactOptionResource,
    false,
    rowsPerPage,
    undefined,
    undefined,
    false,
    true
  )

  const [boostAttributeValueFieldOptionsResponse] = useApiList<IOption<string>>(
    boostAttributeValueFieldOptionsResource,
    false,
    rowsPerPage,
    sourceFieldFixedFilters,
    undefined,
    false,
    true
  )

  if (
    !boostImpactOptionsResponse?.data ||
    !boostAttributeValueFieldOptionsResponse?.data
  ) {
    return null
  }

  return (
    <ProportionalToAttributes
      sourceFields={
        boostAttributeValueFieldOptionsResponse.data['hydra:member']
      }
      boostImpactOptions={boostImpactOptionsResponse.data['hydra:member']}
      {...props}
    />
  )
}

export default ProportionalToAttributesManager
