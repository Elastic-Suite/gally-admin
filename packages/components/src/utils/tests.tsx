import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
