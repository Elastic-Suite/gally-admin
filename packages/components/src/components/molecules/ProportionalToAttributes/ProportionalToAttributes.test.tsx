import React from 'react'

import { renderWithProviders } from '../../../utils/tests'

import ProportionalToAttributes from './ProportionalToAttributes'
import boostImpactOptionsMock from '../../../../public/mocks/boost_impact_options.json'

describe('ProportionalToAttributes', () => {
  it('Should match snapschot', () => {
    const { container } = renderWithProviders(
      <ProportionalToAttributes
        value={{
          source_field_code: undefined,
          boost_impact: undefined,
          scale_factor: 1,
        }}
        sourceFields={[
          {
            id: 'links_exist',
            value: 'links_exist',
            label: 'Product links exist',
          },
          {
            id: 'links_exist2',
            value: 'links_exist2',
            label: 'Product links exist2',
          },
        ]}
        boostImpactOptions={boostImpactOptionsMock['hydra:member']}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
