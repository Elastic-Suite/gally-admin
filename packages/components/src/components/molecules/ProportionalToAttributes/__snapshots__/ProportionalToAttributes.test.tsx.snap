// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProportionalToAttributes Should match snapschot 1`] = `
<div>
  <div
    class="css-1821gv5"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-1wl9gy3-MuiFormControl-root"
    >
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1qqsdnr-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-1w6fxiz-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
          >
            boostConfig.sourceField
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="InputText__Wrapper css-17fp4jk"
          >
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-159ra7o-MuiInputBase-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-ngazp5-MuiInputBase-input"
                id=":r0:"
                placeholder="boostConfig.sourceField.placeholder"
                role="combobox"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="form.open"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="form.open"
                  type="button"
                >
                  <ion-icon
                    src="/images/ionicons/chevron-down.svg"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal css-e5cteh-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        style="position: relative;"
      >
        boostConfig.boostImpact
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
        required=""
        role="radiogroup"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-fbz9fk-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary css-14bw0ew-MuiButtonBase-root-MuiRadio-root"
          >
            <input
              class="PrivateSwitchBase-input css-1m9pwf3"
              name=":r1:"
              type="radio"
              value="log1p"
            />
            <span
              class="css-hyxlzm"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                data-testid="RadioButtonUncheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                data-testid="RadioButtonCheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1kgipl8-MuiTypography-root"
          >
            Bas
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-fbz9fk-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary css-14bw0ew-MuiButtonBase-root-MuiRadio-root"
          >
            <input
              class="PrivateSwitchBase-input css-1m9pwf3"
              name=":r1:"
              type="radio"
              value="sqrt"
            />
            <span
              class="css-hyxlzm"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                data-testid="RadioButtonUncheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                data-testid="RadioButtonCheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1kgipl8-MuiTypography-root"
          >
            Moyen
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-fbz9fk-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary css-14bw0ew-MuiButtonBase-root-MuiRadio-root"
          >
            <input
              class="PrivateSwitchBase-input css-1m9pwf3"
              name=":r1:"
              type="radio"
              value="none"
            />
            <span
              class="css-hyxlzm"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                data-testid="RadioButtonUncheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                data-testid="RadioButtonCheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1kgipl8-MuiTypography-root"
          >
            Haut
          </span>
        </label>
      </div>
    </div>
    <div
      class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
    >
      <div
        class="InputText__Wrapper css-17fp4jk"
      >
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-33c5sc-MuiInputBase-root"
          style="display: none;"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
      >
        boostConfig.scaleFactor
        <span
          aria-label="boostConfig.scaleFactorTooltip"
          class=""
          data-mui-internal-clone-element="true"
          style="display: inline-block; padding-left: 10px; margin-bottom: -5px;"
        >
          <ion-icon
            src="/images/ionicons/information-circle-outline.svg"
            style="color: rgb(129, 135, 185); width: 20px; height: 20px; display: flex;"
          />
        </span>
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="InputText__Wrapper css-17fp4jk"
      >
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-243101-MuiInputBase-root"
          required=""
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
            min="0"
            required=""
            step="any"
            type="number"
            value="1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
